### DISCUSSION TITLE
Enabling "Strict Mode" on Eloquent Model

### DISCUSSION THREAD

**Discussion by simonmaass** [owen-it/laravel-auditing#797] [Q&A] [answered]
When enabling strict mode: https://laravel.com/docs/9.x/eloquent#configuring-eloquent-strictness

`Model::shouldBeStrict(! $this->app->isProduction());`

It will print an error:

`The attribute [auditEvents] either does not exist or was not retrieved for model [App\Models\User].`

![image](https://user-images.githubusercontent.com/32599710/196435509-2abb9e24-eeee-4ff3-adac-a29536e5d5a6.png)

**Comment by nwaweru** [comment]
Experiencing the same.

Any solution?

**Comment by AndreySavenkov** [comment]
Same problem after trying to enable `Model::preventAccessingMissingAttributes()`

**Comment by angeljqv** [comment, marked as answer]
Declare audit properties on User model

```php
public $auditEvents = null;

public $auditDriver = null;

public $auditExclude = null;

public $auditInclude = null;

public $auditStrict = null;

public $auditTimestamps = null;

public $attributeModifiers = null;
```

**Comment by jaseyd** [comment]
> Declare auditEvents on User model
> 
> ```
> public $auditEvents = null;
> ```

Tried this, but same issue.

**Comment by dannypritchard** [comment]
I added the following code to my model and it's working fine:

    public ?array $auditEvents = null;

    public ?string $auditDriver = null;

    public ?array $auditExclude = null;

    public ?array $auditInclude = null;

    public ?bool $auditStrict = null;

    public ?bool $auditTimestamps = null;

    public ?array $attributeModifiers = null;

Might make sense to add this into the `\OwenIt\Auditing\Auditable` trait?

**Comment by simonmaass** [comment]
@anteriovieira ? :)

**Comment by spaantje** [comment]
Any update on this issue?

**Comment by angeljqv** [comment]
> Might make sense to add this into the \OwenIt\Auditing\Auditable trait?

@dannyrevenant  that would be a breaking change, If someone already has any of these properties, they would get a redeclaration exception.

**Comment by MortenDHansen** [comment]
As @angeljqv says, we can't declare those properties in the package. 
I am however not seeing the same issue in my application when enabling strict mode. Can you explain a little more how to reproduce?

**Reply by angeljqv** [reply]
When you have custom changes like on `transformAudit` using those properties, or try to overwrite some methods removing `??`
https://github.com/owen-it/laravel-auditing/blob/5f86086f0465c5701e065771096d0cd9e338a6c9/src/Auditable.php#L531-L566
They are very specific cases that should not be carried out
